cli/cppcheckexecutor.cpp

```diff
@@ -266,7 +281,10 @@ int CppCheckExecutor::check_internal(CppCheck& cppcheck)
         mLatestProgressOutputTime = std::time(nullptr);
 
     if (!settings.outputFile.empty()) {
-        mErrorOutput = new std::ofstream(settings.outputFile);
+        mErrorOutput = new std::ofstream();
+        mErrorOutput->exceptions(std::ios_base::failbit | std::ios_base::badbit);
```

Comment:
Instead of repeating this pattern across we should introduce a helper function for this.